models.go

```diff
@@ -586,3 +586,130 @@ type JSCodeSession struct {
 	UserID     string `json:"userid"`
 	SessionKey string `json:"session_key"`
 }
+
+// MsgAuditEdition 会话内容存档版本
+type MsgAuditEdition uint8
+
+const (
+	// MsgAuditEditionOffice 会话内容存档办公版
+	MsgAuditEditionOffice MsgAuditEdition = 1
+	// MsgAuditEditionService 会话内容存档服务版
+	MsgAuditEditionService MsgAuditEdition = 2
+	// MsgAuditEditionEnterprise 会话内容存档企业版
+	MsgAuditEditionEnterprise MsgAuditEdition = 3
+)
+
+type reqListPermitUser struct {
+	MsgAuditEdition MsgAuditEdition `json:"type"`
+}
+
+var _ bodyer = reqListPermitUser{}
+
+func (x reqListPermitUser) intoBody() ([]byte, error) {
+	result, err := json.Marshal(x)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+type respListPermitUser struct {
+	respCommon
+	IDs []string `json:"ids"`
+}
+
+// CheckSingleAgreeUserInfo 获取会话同意情况（单聊）内外成员
+type CheckSingleAgreeUserInfo struct {
+	UserID         string `json:"userid"`
+	ExternalOpenID string `json:"exteranalopenid"`
+}
+
+// CheckSingleAgreeInfo 获取会话同意情况（单聊）同意信息
+type CheckSingleAgreeInfo struct {
+	StatusChangeTime int    `json:"status_change_time"`
+	AgreeStatus      string `json:"agree_status"`
+	CheckSingleAgreeUserInfo
+}
+
+type reqCheckSingleAgree struct {
+	Infos []CheckSingleAgreeUserInfo `json:"info"`
+}
+
+var _ bodyer = reqCheckSingleAgree{}
+
+func (x reqCheckSingleAgree) intoBody() ([]byte, error) {
+	result, err := json.Marshal(x)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+type respCheckSingleAgree struct {
+	respCommon
+	AgreeInfo []CheckSingleAgreeInfo `json:"agreeinfo"`
+}
+
+// CheckRoomAgreeInfo 获取会话同意情况（群聊）同意信息
+type CheckRoomAgreeInfo struct {
+	StatusChangeTime int    `json:"status_change_time"`
+	AgreeStatus      string `json:"agree_status"`
+	ExternalOpenID   string `json:"exteranalopenid"`
+}
+
+type reqCheckRoomAgree struct {
+	RoomID string `json:"roomid"`
+}
+
+var _ bodyer = reqCheckRoomAgree{}
+
+func (x reqCheckRoomAgree) intoBody() ([]byte, error) {
+	result, err := json.Marshal(x)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+type respCheckRoomAgree struct {
+	respCommon
+	AgreeInfo []CheckRoomAgreeInfo `json:"agreeinfo"`
+}
+
+// GroupChatMember 获取会话内容存档内部群成员
+type GroupChatMember struct {
+	MemberID int `json:"memberid"`
+	JoinTime int `json:"jointime"`
+}
+
+// GroupChat 获取会话内容存档内部群信息
+type GroupChat struct {
+	Members        []GroupChatMember `json:"members"`
+	RoomName       string            `json:"roomname"`
+	Creator        string            `json:"creator"`
+	RoomCreateTime int               `json:"room_create_time"`
```

Comment:
expose this and other timestamp fields as `time.Time` for ergonomic downstream usage? see `media_api.go` for one example of doing the conversion before returning to biz logic.